parser.js

```diff
@@ -766,13 +778,66 @@
         group.name = name;
         return group;
       }
+      else if (features.modifiersGroup && str.indexOf("(?") == pos && str[pos+2] != ":") {
+        return parseModifiersGroup();
+      }
       else {
         //      ( Disjunction )
         //      ( ? : Disjunction )
         return parseGroup('(?:', 'ignore', '(', 'normal');
       }
     }
 
+    function parseModifiersGroup() {
+      function hasDupChar(str) {
+        var i = 0;
+        while (i < str.length) {
+          if (str.indexOf(str[i], i + 1) != -1) {
+            return true;
+          }
+          i++;
+        }
+        return false;
+      }
+
+      var from = pos;
+      incr(2);
+
+      var enablingFlags = matchReg(/^[sim]+/);
+      var disablingFlags;
+      if(match("-")){
+        disablingFlags = matchReg(/^[sim]+/);
+        if (!disablingFlags) {
+          bail('Invalid flags for modifiers group');
+        }
+      } else if(!enablingFlags){
+        bail('Invalid flags for modifiers group');
+      }
+
+      enablingFlags = enablingFlags ? enablingFlags[0] : "";
+      disablingFlags = disablingFlags ? disablingFlags[0] : "";
+
+      var flags = enablingFlags + disablingFlags;
+      if(flags.length > 3 || hasDupChar(flags)){
+        bail('flags cannot be duplicated for modifiers group');
+      }
+
+      var modifiersGroup = {
+        type: 'modifiersGroup',
```

Comment:
Instead of creating a new node, it's enough to add a the two properties to the existing `group` node type. We can consider `(?:` as a special case of this new proposal in every aspect.